tservice/src/seal.rs

```diff
@@ -90,10 +90,74 @@ impl<T: AsRef<[u8]> + core::default::Default> SealedBuilder<T> {
     }
 }
 
+pub trait Unseal: AsRef<[u8]> {
+    /// The length (in bytes) needed to hold the decrypted text
+    fn decrypted_text_len(&self) -> Result<usize> {
+        let result = unsafe {
+            mc_sgx_tservice_sys::sgx_get_encrypt_txt_len(
+                self.as_ref().as_ptr() as *const sgx_sealed_data_t
+            )
+        };
+        match result {
+            // Per the documentation, UINT32_MAX indicates an error
+            u32::MAX => Err(Error::Unexpected),
+            size => Ok(size as usize),
+        }
+    }
+
+    /// Unseal the data in `self`
+    ///
+    /// Returns the unsealed data into the provided `buffer`.  The returned
+    /// slice will be the exact size of the unsealed data.
+    ///
+    /// # Arguments
+    /// * `buffer` - The buffer to output the unsealed data into.  `buffer`
+    ///   needs to be at least as big as [`decrypted_text_len`].
+    fn unseal<'a>(&self, buffer: &'a mut [u8]) -> Result<&'a mut [u8]> {
+        let data_length = self.decrypted_text_len()?;
+        if buffer.len() < data_length {
+            return Err(Error::InvalidParameter);
```

Comment:
It may be worth exercising this codepath in the tests.

Reply:
see above